### PR TITLE
Documentation for Knowledge Distillation

### PR DESCRIPTION
**Pull request by ljaljushkin**

### CODE REVIEW

File: examples/torch/classification/configs/pruning/resnet34_pruning_geometric_median_kd.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    },
+   "compression": [
+       {
+            "algorithm": "filter_pruning",
+            "pruning_init": 0.1,
+            "params": {
+            "schedule": "exponential",
+                "pruning_target": 0.4,
+                "pruning_steps": 15,
+                "weight_importance": "geometric_median"
```
Comment (vshampor) [inline examples/torch/classification/configs/pruning/resnet34_pruning_geometric_median_kd.json:34]: Should read `"filter_importance"`

**Review by vshampor — CHANGES_REQUESTED** [review]

File: docs/compression_algorithms/KnowledgeDistillation.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+### Knowledge Distillation (experimental feature)
+
+#### The algorithm description
+The Knowledge Distillation [Hinton et al., 2015](https://arxiv.org/pdf/1503.02531.pdf)
+implies that a small model (student) is trained to mimic a pre-trained large model (teacher) through knowledge 
+transferring. The goal is to improve the accuracy of the student network.
```
Comment (alexsu52) [inline docs/compression_algorithms/KnowledgeDistillation.md:6]:
```suggestion
transfer. The goal is to improve the accuracy of the student network. 
```

File: examples/torch/classification/configs/pruning/resnet34_pruning_geometric_median_kd.json (hunk 1 above)
Comment (ljaljushkin) [reply examples/torch/classification/configs/pruning/resnet34_pruning_geometric_median_kd.json]: Done

File: docs/compression_algorithms/KnowledgeDistillation.md (hunk 2 above)
Comment (ljaljushkin) [reply docs/compression_algorithms/KnowledgeDistillation.md]: Done

**Review by alexsu52 — APPROVED** [review]

**Review by vshampor — APPROVED** [review]

**Comment by ljaljushkin** [comment]
please merge

**Pull request merged**